feat(publish): add --summary-file option (#2653)
lerna/lerna: commands/publish/README.md

- [`--tag-version-prefix`](#--tag-version-prefix)
- [`--temp-tag`](#--temp-tag)
- [`--yes`](#--yes)

> - [`--summary-file <dir/filename>`](#--summary-file)

Reviewer: Based on the code it doesn't seem like this can be a dir name?

When run with this flag, `lerna publish` will skip all confirmation prompts.
Useful in [Continuous integration (CI)](https://en.wikipedia.org/wiki/Continuous_integration) to automatically answer the publish confirmation prompt.

### `--summary-file`

```sh

> lerna publish --canary --yes --summary-file ./output.json

Reviewer: See comment below on `output.json`, please also make it clear that the value for `--summary-file` is optional and has a default value if not provided

Author: Yeah I've simplified this to only accept a dir and have a default output file name

```

> When run with this flag, once a successfully publish it will create a json summary report(see below for an example).

Reviewer: ```suggestion
When run with this flag, a json summary report will be generated after all packages have been successfully published (see below for an example).
```

When run with this flag, `lerna publish` will skip all confirmation prompts.
Useful in [Continuous integration (CI)](https://en.wikipedia.org/wiki/Continuous_integration) to automatically answer the publish confirmation prompt.

### `--summary-file`

```sh

> # Will create a summary file in the root directory ./lerna-publish-summary.json

Reviewer: ```suggestion
# Will create a summary file in the root directory, i.e. `./lerna-publish-summary.json`
```

lerna publish --canary --yes --summary-file

> # Will create a summary file in the root directory ./some/other/dir/lerna-publish-summary.json

Reviewer: ```suggestion
# Will create a summary file in the provided directory, i.e. `./some/other/dir/lerna-publish-summary.json`
```
